JTCalendar/JTCalendarDayView.m

```diff
@@ -217,27 +224,39 @@ - (void)setSelected:(BOOL)selected animated:(BOOL)animated
     }
     else if([self isToday]){
         if(!self.isOtherMonth){
-            circleView.color = [self.calendarManager.calendarAppearance dayCircleColorToday];
-            textLabel.textColor = [self.calendarManager.calendarAppearance dayTextColorToday];
-            dotView.color = [self.calendarManager.calendarAppearance dayDotColorToday];
+            circleView.color = [calendarAppearance dayCircleColorToday];
+            textLabel.textColor = [calendarAppearance dayTextColorToday];
+            dotView.color = [calendarAppearance dayDotColorToday];
         }
         else{
-            circleView.color = [self.calendarManager.calendarAppearance dayCircleColorTodayOtherMonth];
-            textLabel.textColor = [self.calendarManager.calendarAppearance dayTextColorTodayOtherMonth];
-            dotView.color = [self.calendarManager.calendarAppearance dayDotColorTodayOtherMonth];
+            circleView.color = [calendarAppearance dayCircleColorTodayOtherMonth];
+            textLabel.textColor = [calendarAppearance dayTextColorTodayOtherMonth];
+            dotView.color = [calendarAppearance dayDotColorTodayOtherMonth];
         }
     }
     else{
         if(!self.isOtherMonth){
-            textLabel.textColor = [self.calendarManager.calendarAppearance dayTextColor];
-            dotView.color = [self.calendarManager.calendarAppearance dayDotColor];
+            textLabel.textColor = [calendarAppearance dayTextColor];
+            dotView.color = [calendarAppearance dayDotColor];
         }
         else{
-            textLabel.textColor = [self.calendarManager.calendarAppearance dayTextColorOtherMonth];
-            dotView.color = [self.calendarManager.calendarAppearance dayDotColorOtherMonth];
+            textLabel.textColor = [calendarAppearance dayTextColorOtherMonth];
+            dotView.color = [calendarAppearance dayDotColorOtherMonth];
+        }
+
+        if(self.date && calendarAppearance && calendarAppearance.useCustomDayCircleColor){
```

Comment:
Maybe this could be written differently but I don't know how. It was reaching this line with a nil `date` or `calendarManager` so I put this if statement to filter those situations out.